File: Day11/Peak Index in a Mountain Array/PeakIndexInMountainArray.cs
public class Solution 
{
    public int PeakIndexInMountainArray(int[] arr) 
    {
        int beg = 0, end = arr.Length - 1;
        int mid = 0;

        while(beg <= end)
        {
            mid = beg + ((end - beg) / 2);

            if(arr[mid] > arr[mid + 1] && arr[mid] > arr[mid - 1])
            {
                return mid;
            }
            else if(arr[mid] < arr[mid + 1])
            {
                beg = mid + 1;
            }
            else if(arr[mid] > arr[mid + 1])
            {
                end = mid - 1;
            }
        }

        return mid;    
    }
}

File: Day11/Peak Index in a Mountain Array/PeakIndexInMountainArray.py
class Solution:
    def peakIndexInMountainArray(self, arr) -> int:
        beg = 0
        end = len(arr) - 1
        mid = 0
        
        while beg <= end:
            mid = beg = ((beg + end)// 2)
            
            if(arr[mid - 1] < arr[mid] > arr[mid + 1]):
                return mid
            elif (arr[mid - 1] < arr[mid]):
                beg = mid + 1
            elif (arr[mid - 1] > arr[mid]):
                end = mid - 1
                
        return mid
        